e2e/nomostest/git.go

```diff
@@ -501,7 +506,35 @@ func (g *Repository) CommitAndPushBranch(msg, branch string) {
 	g.Git("commit", "-m", msg)
 
 	g.Logger.Infof("[repo %s] committing %q (%s)", path.Base(g.Root), msg, g.Hash())
-	g.Git("push", "-f", "-u", remoteOrigin, branch)
+	g.Push(branch, "-f")
+}
+
+// Push pushes the provided refspec to the git server.
+// Performs a retry using RemoteURL, which may change if the port forwarding restarts.
+func (g *Repository) Push(refspec string, flags ...string) {
+	var out []byte
+	took, err := retry.Retry(1*time.Minute, func() error {
+		args := []string{"push"}
+		args = append(args, flags...)
+		args = append(args, g.RemoteURL, refspec)
+		cmd := g.gitCmd(args...)
+		var err error
+		out, err = cmd.CombinedOutput()
+		return err
+	})
+	if err != nil {
+		g.T.Log(string(out))
+		g.T.Fatalf("took %v, err %v", took, err)
+	}
+}
+
+func (g *Repository) pushAllToRemote(remote string) {
+	cmd := g.gitCmd("push", remote, "--all")
```

Comment:
The distinction between Git and gitCmd are confusing, but predate this PR. In the future, we probably want to rename Git to like MustGit or something, since it fails and doesn't return the output. Then gitCmd could be named Git and return the output.

Anyway, it's fine for now, just a note.